src/include/kdb.h.in

```diff
@@ -104,16 +110,19 @@ typedef struct _KeySet	KeySet;
  *
  **************************************/
 
-KDB * kdbOpen(Key *errorKey);
+KDB * kdbOpen(const KeySet * contract, Key *parentKey);
+
 int kdbClose(KDB *handle, Key *errorKey);
 
 int kdbGet(KDB *handle, KeySet *returned,
 	Key *parentKey);
 int kdbSet(KDB *handle, KeySet *returned,
 	Key *parentKey);
 
-int kdbEnsure (KDB * handle, KeySet * contract, Key * parentKey);
-
+int elektraGOptsContract (KeySet * contract, int argc, const char * const * argv, const char * const * envp, const Key * parentKey,
+			  KeySet * goptsConfig);
+int elektraGOptsContractFromStrings (KeySet * contract, size_t argsSize, const char * args, size_t envSize, const char * env,
```

Comment:
Basically all of them, since the lifetime requirements for the pointer arguments of `elektraGOptsContract ` are more or less only possible to fulfil with manual memory management.

Reply:
Btw. if you now say it might also be useful for C users, we should definitely document the (memory management) differences.

Reply:
The documentation for `elektraGOptsContract` quite clearly says: "IMPORTANT: The pointer and data behind must be valid after kdbClose() is called."

https://github.com/ElektraInitiative/libelektra/blob/d3fdba3c36642de3f883dc182e9b9dda377d7a67/src/libs/elektra/contracts.c#L20-L25

Reply:
I do not understand. How does this explains the difference between elektraGOptsContract andelektraGOptsContractFromStrings ?

Reply:
It explains the memory management differences. The other differences should also be obvious from the doxygen comments.

https://github.com/ElektraInitiative/libelektra/blob/d3fdba3c36642de3f883dc182e9b9dda377d7a67/src/libs/elektra/contracts.c#L19-L25

vs

https://github.com/ElektraInitiative/libelektra/blob/d3fdba3c36642de3f883dc182e9b9dda377d7a67/src/libs/elektra/contracts.c#L76-L85